### PR TITLE
report(redesign): inline display text. bold audit header text

### PR DESCRIPTION
**Pull request by Hoten**
#8185

image

### CODE REVIEW

File: lighthouse-core/report/html/report-styles.css [hunk 1]
```diff
@@ -387,14 +386,20 @@
...
 }
 
-.lh-audit__title {
+.lh-audit__display-text {
   flex: 1;
 }
 
+/* Prepend display text with separator. But not in Opportunities. */
+.lh-audit-group:not(.lh-audit-group--load-opportunities) .lh-audit__display-text:not(:empty):before {
+  content: '—';
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]:
```suggestion
  content: '\2014';
```

File: lighthouse-core/report/html/report-styles.css [hunk 2]
```diff
@@ -387,14 +386,20 @@
   margin-right: 0;
 }
 
-.lh-audit__title {
+.lh-audit__display-text {
   flex: 1;
 }
 
+/* Prepend display text with separator. But not in Opportunities. */
+.lh-audit-group:not(.lh-audit-group--load-opportunities) .lh-audit__display-text:not(:empty):before {
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: while this selector does look crazy, it turns out to be reasonable..  👍

File: lighthouse-core/report/html/report-styles.css [hunk 3]
```diff
@@ -387,14 +386,20 @@
   margin-right: 0;
 }
 
-.lh-audit__title {
+.lh-audit__display-text {
   flex: 1;
 }
 
+/* Prepend display text with separator. But not in Opportunities. */
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]:
```suggestion
/* Prepend display text with em dash separator. But not in Opportunities. */
```

File: lighthouse-core/report/html/report-styles.css (hunk 1 above)
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: per discussion in chat.. let's use whitespace on both sides of the dash.

Assuming we want to use `var(--audit-item-gap)` the we'll need some margin in here too:

```cs
  margin-right: var(--audit-item-gap);
```

**Comment by brendankenny** [comment]
these things realllllly need a screenshot. No idea what "inline display text" means here :)

File: lighthouse-core/report/html/report-styles.css (hunk 1 above)
Comment (brendankenny) [reply lighthouse-core/report/html/report-styles.css]: can this just go in the HTML? Kind of weird to do it with a pseudo-element

Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: why? dt encoding issues are resolved

Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: @brendankenny can't just be HTML, need some sort of logic check. makes sense to me to have the display + logic in one place here.

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: :)

File: lighthouse-core/report/html/report-styles.css (hunk 1 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: done

File: lighthouse-core/report/html/report-styles.css [hunk 4]
```diff
@@ -453,13 +452,20 @@
   margin-right: 0;
 }
 
-.lh-audit__title {
+.lh-audit__display-text {
   flex: 1;
 }
 
+/* Prepend display text with em dash separator. But not in Opportunities. */
+.lh-audit-group:not(.lh-audit-group--load-opportunities) .lh-audit__display-text:not(:empty):before {
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: best practices aren't grouped so aren't getting the `:before`

Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: fixed

File: lighthouse-core/report/html/report-styles.css (hunk 1 above)
Comment (brendankenny) [reply lighthouse-core/report/html/report-styles.css]: > can't just be HTML, need some sort of logic check. makes sense to me to have the display + logic in one place here.

I don't think it needs to be. Opportunities and the rest of the audits share text with a `.lh-audit__display-text` class, but they actually come from different templates, `#tmpl-lh-opportunity` vs `#tmpl-lh-audit`. If the `#tmpl-lh-audit` template gets the em dash inline, then it'll only show up there

Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: Good catch on the template thing. But would still need a length check - and can't make a CSS rule for preceding sibling (pseudo code below).

image

```
.lh-audit__separator {
  margin: 0 var(--audit-item-gap);
}
/* not actually a thing */
.lh-audit__display-text:not(:empty) - .lh-audit__separator {
  display: none;
}
```

**Pull request merged**